src/Moq/Mock.Generic.cs

```diff
@@ -164,7 +166,15 @@ private void CheckParameters()
 		public override bool CallBase
 		{
 			get => this.callBase;
-			set => this.callBase = value;
+			set
+			{
+				if (typeof(Delegate).IsAssignableFrom(this.MockedType))
+				{
+					throw new NotSupportedException(string.Format(CultureInfo.CurrentCulture, Resources.CallBaseUsedOnDelegateException));
```

Comment:
It is important that an exception be thrown only if `value == true`. This is because when you have a mock with `DefaultValue.Mock`, and Moq auto-mocks something (e.g. the delegate mock's return value), it'll attempt to set `CallBase`.